infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -132,29 +133,32 @@ public LogUnitServer(ServerContext serverContext) {
     /**
      * Returns a new LogUnitServer.
      *
-     * @param serverContext      context object providing settings and objects
-     * @param serverInitializer  a LogUnitServerInitializer object used for initializing the
-     *                           cache, stream log, and batch processor for this server
+     * @param serverContext     context object providing settings and objects
+     * @param serverInitializer a LogUnitServerInitializer object used for initializing the
+     *                          cache, stream log, and batch processor for this server
      */
     public LogUnitServer(ServerContext serverContext, LogUnitServerInitializer serverInitializer) {
         this.serverContext = serverContext;
         config = LogUnitServerConfig.parse(serverContext.getServerConfig());
         executor = serverContext.getExecutorService(serverContext.getLogUnitThreadCount(), "LogUnit-");
         HealthMonitor.reportIssue(Issue.createInitIssue(Component.LOG_UNIT));
+
+        BatchProcessorContext batchProcessorContext = new BatchProcessorContext();
```

Comment:
`BatchProcessorContext` can be moved to the LogUnitServerInitializer as a field and then it can just be passed directly to the StreamLogs and BatchProcessor constructors without being exposed in the `LogUnitServer` directly.

Reply:
The purpose of that class was just to facilitate mocking by allowing all of the dependencies to be passed via the constructor. This can be circumvented with MockedConstruction/PowerMock, but these were not available in earlier versions of Mockito.